chrome/browser/android/adblock/adblock_bridge.cc

```diff
@@ -84,6 +84,32 @@ std::string replaceString(
     return subject;
 }
 
+std::string escapeJavaScriptString(const std::string& subject) {
```

Comment:
taken from https://github.com/joliss/js-string-escape/blob/master/index.js
and not taking into account utf-16 characters.

Reply:
@xxxz: i'm not aware of UTF details in C++ like special "widechar" string and all this stuff. If you feel we need it i'd ask somebody else to contribute here. BTW since in Chromium it's still `std::string` used i wonder how it works now for utf-16 characters

Reply:
here we iterate over single characters. If we want to check/replace multibyte characters we need just replace strings, not characters. So basically we need to change the whole function

Reply:
yes, one can do it by using index-based loop

Reply:
I would just check for \u2028 and \u2029 (their utf-8 form) to properly escape them and leave any other non-ASCII alone. As mentioned above this mean not using a range iterator but an index based iterator.